scripts/earthengine/events_pipeline.py

```diff
@@ -0,0 +1,818 @@
+                    filename = f'{output_dir}/{import_name}-{stage_name}-{start_date}-{time_period}{file_ext}'
+
+        return _format(filename, self.config)
+
+    def should_skip_existing_output(self, filename: str) -> bool:
+        '''Returns True if the filename exists and skip_existing_output is True.'''
+        if self.get_config('skip_existing_output', True):
+            existing_file = file_util.file_get_matching(filename)
+            if existing_file:
+                return True
+        return False
+
+    def get_config(self,
+                   param: str,
+                   default_value: str = '',
+                   config_dict: dict = {}) -> str:
+        '''Returns the resolved config value from config_dict or object's config.'''
+        if config_dict:
+            value = config_dict.get(param, None)
+            if value:
+                return _format(value, config_dict)
+        value = self.config.get(param, default_value)
+        return _format(value, self.config)
+
+    def set_config(self, param: str, value: str) -> str:
+        '''Set the config parameter.'''
+        self.config[param] = value
+        return value
+
+    def get_configs(self) -> dict:
+        return _get_resolved_dict(self.config)
+
+    def get_state(self, key: str, default_value: str = '') -> str:
+        '''Returns the value for the key from the processing state dict.'''
+        return self.state.get(key, default_value)
+
+    def set_state(self, key: str, value: str):
+        '''Sets the value of the pipeline state.'''
+        self.state[key] = value
+        return value
+
+    def set_config_dates(self, start_date: str = ''):
+        '''Set the dates in the config based on processing state.'''
+        # Set start_date to be the next date from the last input date
+        if start_date:
+            self.set_config('start_date', start_date)
+        start_date = self.get_config('start_date', '')
+        if not start_date:
+            # Set the start_date to the last processed date.
+            last_input_date = self.get_state('last_input_date', '')
+            if last_input_date:
+                start_date = utils.date_advance_by_period(
+                    self.get_state('last_input_date', ''),
+                    self.get_config('time_period', 'P1M'))
+                logging.info(
+                    f'Setting start_date to {start_date} from last_input_date: {last_input_date}'
+                )
+            if not start_date:
+                # No date set, set to start of year.
+                start_date = utils.date_today('%Y-01-01')
+                logging.info(
+                    f'Setting start_date to start of year: {start_date}')
+            if start_date:
+                self.set_config('start_date', start_date)
+        self.set_config('year', start_date[:4])
+        self.set_config('month', start_date[5:7])
+        self.set_config('day', start_date[8:10])
+
+
+class EarthEngineRunner(StageRunner):
```

Comment:
nit: perhaps in a follow-up PR, should each of these specific Runners be in their own file?

Reply:
+1

Reply:
Split runners into separate files pipeline_stage_*.py
test is a single file though for now, events_pipeline_test.py tests multiple stages.